pkg/portal/graphql/app_mutation.go

```diff
@@ -69,30 +71,35 @@ var _ = registerMutationField(
 
 			gqlCtx := GQLContext(p.Context)
 
-			lazy := gqlCtx.Apps.Get(appID)
-			return lazy.
-				Map(func(value interface{}) (interface{}, error) {
-					app := value.(*model.App)
-					var resourceUpdates []resources.Update
-					for _, f := range updates {
-						f := f.(map[string]interface{})
-						path := f["path"].(string)
-						var data []byte
-						if stringData, ok := f["data"].(string); ok {
-							data = []byte(stringData)
-						}
-
-						resourceUpdates = append(resourceUpdates, resources.Update{
-							Path: path,
-							Data: data,
-						})
-					}
-
-					return map[string]interface{}{
-						"app": gqlCtx.Apps.UpdateResources(app, resourceUpdates),
-					}, nil
-				}).
-				Value, nil
+			app, err := gqlCtx.AppService.Get(appID)
+			if err != nil {
+				return nil, err
+			}
+
+			var resourceUpdates []resources.Update
+			for _, f := range updates {
+				f := f.(map[string]interface{})
+				path := f["path"].(string)
+				var data []byte
+				if stringData, ok := f["data"].(string); ok {
+					data = []byte(stringData)
+				}
+
+				resourceUpdates = append(resourceUpdates, resources.Update{
+					Path: path,
+					Data: data,
+				})
+			}
+
+			err = gqlCtx.AppService.UpdateResources(app, resourceUpdates)
+			if err != nil {
+				return nil, err
+			}
+
+			// App is not primed here intentionally.
```

Comment:
Any reason behind this?

Reply:
I meant the app from line 74 was not primed because this mutation mutates app. So it is safer to not prime it.